libparsec/crates/platform_mountpoint/tests/base.rs

```diff
@@ -0,0 +1,260 @@
+
+use std::fs::{File, OpenOptions};
+
+use libparsec_platform_mountpoint::mount;
+use libparsec_tests_fixtures::prelude::*;
+
+macro_rules! mount {
+    ($tmp_path: ident) => {{
+        #[cfg(target_os = "windows")]
+        winfsp_wrs::init().expect("Can't init WinFSP");
+
+        // Windows can't mount on existing directory
+        let path = (*$tmp_path).join("mountpoint");
+        let fs = mount(&path).unwrap();
+
+        (path, fs)
+    }};
+}
+
+#[parsec_test]
+fn create_dir(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+
+    std::fs::create_dir(&bar).unwrap();
+
+    assert!(bar.exists());
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn create_file(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+
+    File::create(&bar).unwrap();
+
+    assert!(bar.exists());
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn stop(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+
+    std::fs::create_dir(&bar).unwrap();
+
+    fs.stop();
+
+    // unmount should remove all its content
+    assert!(!bar.exists());
+}
+
+#[parsec_test]
+#[case("x".repeat(256))]
+#[case("飞".repeat(85) + "x")]
+fn dir_exceeding_255_bytes(tmp_path: TmpPath, #[case] name: String) {
+    let (path, fs) = mount!(tmp_path);
+
+    // TODO: assert_eq it
+    // ErrorKind::InvalidFilename is under `io_error_more` feature which is
+    // available on nightly only
+    std::fs::create_dir(path.join(name)).unwrap_err();
+
+    fs.stop();
+}
+
+#[parsec_test]
+#[case("x".repeat(256))]
+#[case("飞".repeat(85) + "x")]
+fn file_exceeding_255_bytes(tmp_path: TmpPath, #[case] name: String) {
+    let (path, fs) = mount!(tmp_path);
+
+    // TODO: assert_eq it
+    // ErrorKind::InvalidFilename is under `io_error_more` feature which is
+    // available on nightly only
+    File::create(path.join(name)).unwrap_err();
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn read_write(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let data = b"Lorem ipsum";
+
+    File::create(&bar).unwrap();
+
+    std::fs::write(&bar, data).unwrap();
+
+    p_assert_eq!(std::fs::read(&bar).unwrap(), data);
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn remove_dir(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+
+    std::fs::create_dir(&bar).unwrap();
+    std::fs::remove_dir(&bar).unwrap();
+
+    assert!(!bar.exists());
+
+    fs.stop();
+}
+
+#[parsec_test]
+fn remove_dir_non_empty(tmp_path: TmpPath) {
+    let (path, fs) = mount!(tmp_path);
+    let bar = path.join("bar");
+    let foo = bar.join("foo");
+
+    std::fs::create_dir(&bar).unwrap();
+    std::fs::create_dir(foo).unwrap();
+
+    // TODO: assert_eq it
+    // ErrorKind::DirectoryNotEmpty is under `io_error_more` feature which is
+    // available on nightly only
+    std::fs::remove_dir(&bar).unwrap_err();
```

Comment:
```suggestion
    std::fs::remove_dir(&bar).unwrap_err();
    
    std::fs::remove_dir_all(&bar).unwrap();
```

You can also check that `remove_dir_all` work